### ISSUE TITLE
Minimum distances are not respected

### ISSUE THREAD

**Issue by jirilukavsky** [jirilukavsky/motrack#7]
Positions and trajectories with minimum distance > 1 often violate minimum distance requirement:
```
library(tidyverse)
library(motrack)

a <- 12
diameter_object <- 2
sett_move <-
  new_settings(speed = 6, 
               xlim = c(-a, a), ylim = c(-a, a),
               bounce_off_square = F, 
               min_dist = diameter_object * 1.5, 
               r = diameter_object / 2,
               arena_shape = "circle",
               bounce_off_circle = T, circle_bounce_jitter = pi / 6)

set.seed(1001)
position <- generate_positions_random(10, sett_move)
plot_position(position, sett_show)
```

**Comment by jirilukavsky** [comment]
The bug is likely to be resolved now, but it would be nice to have unit tests for this. 
Probably with some tolerance because bouncing does not check for distances after it corrects directions. With large time steps between frames, it is possible to violate `min_distance` in this manner.

**Comment by jirilukavsky** [comment]
Closed, test unit opened in separate issue #9
